### ISSUE TITLE
Oracle RDS character_set_name is not being respected on Read Replicas

### ISSUE THREAD

**Issue by w0jnar** [hashicorp/terraform-provider-aws#20379]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

* Terraform v0.14.11
* hashicorp/aws v3.52.0

### Affected Resource(s)

* aws_db_instance

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_db_instance" "oracle" {
  name                                = "TEST01"
  allocated_storage                   = 100
  storage_type                        = "gp2"
  instance_class                      = "db.m5.large"
  engine                              = "oracle-ee"
  engine_version                      = "19.0.0.0.ru-2020-10.rur-2020-10.r1"
  license_model                       = "bring-your-own-license"
  password                            = "XXX"
  username                            = "XXX"
  replicate_source_db                 = null
  character_set_name                  = "WE8ISO8859P15"
}

resource "aws_db_instance" "oracle_replica" {
  name                                = null
  allocated_storage                   = 100
  storage_type                        = "gp2"
  instance_class                      = "db.m5.large"
  engine                              = "oracle-ee"
  engine_version                      = "19.0.0.0.ru-2020-10.rur-2020-10.r1"
  license_model                       = "bring-your-own-license"
  replicate_source_db                 = aws_db_instance.oracle.arn
  character_set_name                  = "WE8ISO8859P15"
}

resource "aws_db_instance" "oracle_replica_cross_region" {
  provider = aws.secondary

  name                                = null
  allocated_storage                   = 100
  storage_type                        = "gp2"
  instance_class                      = "db.m5.large"
  engine                              = "oracle-ee"
  engine_version                      = "19.0.0.0.ru-2020-10.rur-2020-10.r1"
  license_model                       = "bring-your-own-license"
  replicate_source_db                 = aws_db_instance.oracle.arn
  character_set_name                  = "WE8ISO8859P15"
}
```

### Expected Behavior

When creating a read replica of this instance, where the `character_set_name = "WE8ISO8859P15"`, we'd expect the replica to have the same `character_set_name` set and this shows up in the plan correctly if `character_set_name = "WE8ISO8859P15"` was set for the replica as well.

### Actual Behavior

If the read replica is created in the same region as the primary, the replica copies the `character_set_name` of the primary, ignoring what was in the plan or provided, and it cannot be modified.
If a read replica is created in a different region (in our case, the primary was in eu-west-1, the read replica was being created in eu-central-1), the read replica gets the default `AL32UTF8` regardless of what the primary was set to, or if we provided it as a parameter to the replica. The plan again shows the value that was set as a parameter.

In any case, I'd expect some kind of consistency, either the read replica copying the same character set by default from the primary regardless of region or allowing the user to set the value at creation seeing as it forces replacement. Right now we have different behaviors depending on regions and plans showing inconsistent behavior.

### Steps to Reproduce

1. Create an Oracle RDS instance with a non-default character set (ex: `WE8ISO8859P15`)
2. Create a read replica in another region from the instance created from step 1.
3. Verify the `character_set_name` on the read replica is `AL32UTF8` rather than what it was set.

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/issues/31719.
Relates https://github.com/hashicorp/terraform-provider-aws/pull/36080.

**Comment by gdavison** [comment]
`character_set_name` cannot be set when creating a replica.

There appears to be an error on the AWS side when creating a cross-region replica. The character set should be set by AWS.

The PR #36518 will add a warning when `character_set_name` is set on a replica.

**Comment by gdavison** [comment]
Closed by #36518
